File: app/src/main/java/fergaral/datetophoto/DateToPhoto.java
package fergaral.datetophoto;

import android.app.Application;
import android.widget.Toast;

import com.evernote.android.job.JobManager;

import fergaral.datetophoto.jobs.MyJobCreator;
import fergaral.datetophoto.jobs.ProcessPhotosJob;

/**
 * Created by Fer on 07/10/2017.
 */

public class DateToPhoto extends Application {

    private static DateToPhoto instance;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        JobManager.create(this).addJobCreator(new MyJobCreator());
        ProcessPhotosJob.scheduleNow();
    }

    public static DateToPhoto getInstance() {
        return instance;
    }
}

File: app/src/main/java/fergaral/datetophoto/DateToPhoto.kt
package fergaral.datetophoto

import android.app.Application
import android.preference.PreferenceManager
import androidx.appcompat.app.AppCompatDelegate
import fergaral.datetophoto.utils.ThemeUtils

import fergaral.datetophoto.works.ProcessPhotosWorker

/**
 * Created by Fer on 07/10/2017.
 */

class DateToPhoto : Application() {

    override fun onCreate() {
        super.onCreate()
        instance = this
        ProcessPhotosWorker.schedule()
        readTheme()
    }

    private fun readTheme() {
        val themeValue = PreferenceManager.getDefaultSharedPreferences(this)
                .getString(getString(R.string.pref_theme_key), getString(R.string.pref_theme_default)) as String
        ThemeUtils.changeTheme(themeValue)
    }

    companion object {
        lateinit var instance: DateToPhoto
            private set
    }
}
